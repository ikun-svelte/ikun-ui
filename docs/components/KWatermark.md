---
title: KWatermark
lang: en-US
---

# KWatermark

Add text or images to pages to protect page content.

## Install

::: code-group

```bash [pnpm]
pnpm add @ikun-ui/watermark
```

```bash [yarn]
yarn add @ikun-ui/watermark
```

```bash [npm]
npm install @ikun-ui/watermark
```

:::

## Basic usage

<demo src="watermark/basic.svelte"  github='Watermark'></demo>

## Multi-line

Use `content` to set an array of strings to specify multi-line text watermark content.

<demo src="watermark/content.svelte" github='Watermark'></demo>

## Image

Specify the image address via `image`.
To ensure that the image is clearly displayed and not stretched, please set the width and height.
It is recommended to use an image with at least twice the width and height to ensure the display effect.

<demo src="watermark/image.svelte" github='Watermark'></demo>

## Watermark Props

| Name    | Type                 | Default                | Description                                                                                      |
| ------- | -------------------- | ---------------------- | ------------------------------------------------------------------------------------------------ |
| width   | `number`             | `120`                  | The width of the watermark, the default value of `content` is its own width                      |
| height  | `number`             | `64`                   | The height of the watermark, the default value of `content` is its own height                    |
| rotate  | `number`             | `-22`                  | When the watermark is drawn, the rotation Angle, unit `Â°`                                        |
| zIndex  | `number`             | `9`                    | The z-index of the appended watermark element                                                    |
| image   | `string`             | `-`                    | Image source, it is recommended to export 2x or 3x image, high priority.                         |
| content | `string[] \| string` | `-`                    | Watermark text content.                                                                          |
| font    | `Font`               | `Font`                 | Text style.                                                                                      |
| gap     | `[number, number]`   | `[100, 100]`           | The spacing between watermarks.                                                                  |
| offset  | `[number, number]`   | `[gap[0]/2, gap[1]/2]` | The offset of the watermark from the upper left corner of the container. The default is `gap/2`. |
| cls     | `string`             | `''`                   | Additional class for component                                                                   |
| attrs   | `any`                | `{}`                   | Additional attributes                                                                            |

## Watermark Slots

| Name    | Description               |
| ------- | ------------------------- |
| default | content of the watermark. |
