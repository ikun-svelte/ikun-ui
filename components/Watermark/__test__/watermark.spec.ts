import { afterEach, expect, test, describe, beforeEach } from 'vitest';
import KWatermark from '../src';
import KWatermarkSlot from './fixture/watermark.slot.svelte';
let host: HTMLElement;

const initHost = () => {
	host = document.createElement('div');
	host.setAttribute('id', 'host');
	document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
});
afterEach(() => {
	host.remove();
});

describe('Test: KWatermark', () => {
	test('props: cls', async () => {
		const instance = new KWatermark({
			target: host,
			props: {
				cls: 'k-watermark--test'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('k-watermark--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('slot: default', async () => {
		const instance = new KWatermarkSlot({
			target: host
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('watermark_slot')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});
});
