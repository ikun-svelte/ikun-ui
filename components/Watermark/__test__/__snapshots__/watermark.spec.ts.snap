// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KWatermark > props: cls 1`] = `"<div class=\\"k-watermark k-watermark--base k-watermark--test\\" style=\\"position: relative;\\"></div>"`;

exports[`Test: KWatermark > slot: default 1`] = `" <div class=\\"k-watermark k-watermark--base\\" style=\\"position: relative;\\"><div style=\\"height: 500px;\\" id=\\"watermark_slot\\"></div></div>"`;
